tests/rest/client/test_relations.py

```diff
@@ -982,7 +982,7 @@ def assert_annotations(bundled_aggregations: JsonDict) -> None:
                 bundled_aggregations,
             )
 
-        self._test_bundled_aggregations(RelationTypes.ANNOTATION, assert_annotations, 7)
+        self._test_bundled_aggregations(RelationTypes.ANNOTATION, assert_annotations, 6)
```

Comment:
This (and the two other similar changes) is because these tests work by counting database transactions. In these transactions there would be `get_retention_policy_for_room` because we would always run it when responding with events. This PR changes things so that we skip it if retention is disabled, which mean it's one less transaction happening.